testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeleteAccountTest.java

```diff
@@ -0,0 +1,210 @@
+    super.navigateBeforeTest();
+  }
+
+  @Override
+  @Test
+  public void navigationTest() {
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles();
+    driver.navigate().refresh();
+    super.navigationTest();
+    disableDeleteAccountRequiredAction();
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountPageNotVisibleAndNotAccessibleWithoutUserRole() {
+    enableDeleteAccountRequiredAction();
+    List<WebElement> accountLinks = driver.findElements(By.cssSelector("div[id^=\"landing-\"]"));
+   Assert.assertTrue(accountLinks.stream().noneMatch(link -> link.getAttribute("id").contains("delete-account")));
+    deleteAccountPage.navigateTo();
+    pageNotFound.assertCurrent();
+    //reset role back since realm is shared among tests
+    disableDeleteAccountRequiredAction();
+  }
+
+
+  @Test
+  public void deleteOwnAccountPageNotVisibleAndNotAccessibleWithoutDeleteAccountActionEnabled() {
+    addDeleteAccountRoleToUserClientRoles();
+    List<WebElement>  accountLinks = driver.findElements(By.cssSelector("div[id^=\"landing-\"]"));
+    Assert.assertTrue(accountLinks.stream().noneMatch(link -> link.getAttribute("id").contains("delete-account")));
+    deleteAccountPage.navigateTo();
+    pageNotFound.assertCurrent();
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountAIACancellationSucceeds() {
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles();
+    deleteAccountPage.navigateTo();
+    deleteAccountPage.getDeleteAccountButton().click();
+    loginToAccount();
+    URI uri = URI.create(driver.getCurrentUrl());
+    Assert.assertTrue(uri.getPath().contains("login-actions"));
+    Assert.assertTrue(uri.getQuery().contains("execution=delete_account"));
+    driver.findElement(By.cssSelector("button[name='cancel-aia']")).click();
+    Assert.assertTrue(deleteAccountPage.isCurrent());
+    disableDeleteAccountRequiredAction();
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountForbiddenWithoutDeleteAccountActionEnabled() {
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles();
+    deleteAccountPage.navigateTo();
+    deleteAccountPage.getDeleteAccountButton().click();
+    loginToAccount();
+    disableDeleteAccountRequiredAction();
+    driver.findElement(By.cssSelector("input[type='submit']")).click();
+    Assert.assertEquals(driver.findElements(By.cssSelector(".alert-error")).size(), 1);
+    Assert.assertEquals(driver.findElement(By.cssSelector("#kc-content-wrapper > div > span.kc-feedback-text")).getText(), "You do not have enough permissions to delete your own account, contact admin.");
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountSucceeds() {
+    String userId = createUserToBeDeleted("test-user-to-be-deleted@localhost", "password");
```

Comment:
Why do you need to create another user? The test user is removed at the end of the test anyway, so should be safe to remove it.

Reply:
:+1: